Parameterise `Certificate` by `era`

newhoggy opened the pull request:
# Description

Parameterise `Certificate` by `era`

# Changelog

```yaml
- description: |
    Parameterise `Certificate` by `era`
  compatibility: breaking
  type: feature
```

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

mzabaluev commented:
Fixes #50.

Jimbo4350 committed:
Implement Cardano.CLI.Conway.Parsers
Move more parsers in to Cardano.CLI.Common.Parsers

Jimbo4350 committed:
Implement ConwayProposal type
Move more parsers to Cardano.CLI.ConwayParsers

Jimbo4350 committed:
Wired up create-vote and create-constitution commands

Jimbo4350 reviewed (changes requested):
https://github.com/input-output-hk/cardano-cli/pull/45/commits needs to be merged first

Jimbo4350 committed:
REMOVE ME: Point at the next cardano-api release until it is published
to CHaP

Jimbo4350 committed:
Implement
- Cardano.CLI.Conway.Commands/Types
- ConwayVote
Propagate new TxBodyContent votes and proposals records

disassembler committed:
add missing pattern matches

disassembler committed:
cardano-api: update to 8.7

newhoggy committed:
Parameterise Certificate with era
